server/api/src/rsModColl.cpp

```diff
@@ -73,8 +74,24 @@ namespace
 auto rsModColl(rsComm_t* rsComm, collInp_t* modCollInp) -> int
 {
     namespace ic = irods::experimental::catalog;
+    namespace fs = irods::experimental::filesystem;
+
+    if (!rsComm || !modCollInp) {
+        return SYS_INTERNAL_NULL_INPUT_ERR;
+    }
 
     try {
+        // If the zone name cannot be derived from the collection name in the input, the path cannot be valid.
+        const auto zone_name = fs::zone_name(fs::path{static_cast<char*>(modCollInp->collName)});
+        if (!zone_name) {
+            return SYS_INVALID_FILE_PATH;
+        }
+
+        if (!isLocalZone(zone_name->data())) {
```

Comment:
Let's use `c_str()` instead of `data()`.

Reply:
Done